website/source/docs/job-specification/template.html.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: "docs"
+page_title: "template Stanza - Job Specification"
+sidebar_current: "docs-job-specification-template"
+description: |-
+  The "template" block instantiates an instance of a template renderer. This
+  creates a convenient way to ship configuration files that are populated from
+  Consul data, Vault secrets, or just general configurations within a Nomad
+  task.
+---
+
+# `template` Stanza
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th width="120">Placement</th>
+    <td>
+      <code>job -> group -> task -> **template**</code>
+    </td>
+  </tr>
+</table>
+
+The `template` block instantiates an instance of a template renderer. This
+creates a convenient way to ship configuration files that are populated from
+Consul data, Vault secrets, or just general configurations within a Nomad task.
+
+```hcl
+job "docs" {
+  group "example" {
+    task "server" {
+      template {
+        source        = "local/redis.conf.tpl"
+        destination   = "local/redis.conf"
+        change_mode   = "signal"
+        change_signal = "SIGNINT"
+      }
+    }
+  }
+}
+```
+
+Nomad is utilizes a tool called [Consul Template][ct]. For a full list of the
+API template functions, please see the [Consul Template README][ct].
+
+## `template` Parameters
+
+- `source` `(string: "")` - Specifies the path to the template to be rendered.
```

Comment:
That's a good question. @dadgar could you weigh in here? Is it possible to package the template inside a container, for example, and then reference that as the source?